core/driver/sources/src/main/scala/org/virtuslab/ideprobe/ide/intellij/InstalledIntelliJ.scala

```diff
@@ -219,5 +219,20 @@ final class DownloadedIntelliJ(
   override val ideaProperties: Path =
     root.resolve("bin").resolve("idea.properties").write(ideaPropertiesContent)
 
-  override def cleanup(): Unit = root.delete()
+  override def cleanup(): Unit = {
+    probePaths.diagnostics match {
+      case Some(path) =>
+        root
+          .resolve("logs")
+          .copyDir(path.resolve(getPathWithVersionNumber(root).getFileName).resolve("logs"))
+        root.delete()
+      case None =>
+        root.delete()
+    }
+    root.delete()
+  }
+
+  private def getPathWithVersionNumber(intellijRootPath: Path): Path =
+    if (intellijRootPath.getFileName.toString == "Contents") intellijRootPath.getParent else intellijRootPath
```

Comment:
Hmmm what special case is handled here? 🤔

Reply:
Also, it's not clear why it's a path **with version number**

Reply:
If I get it right, there is a difference:
- for Linux based OS `root` will be something like `intellij-instance-2022.2.1--T3ySdgShSvyr87HNoJq-oQ`
- for macOs `root` will be something like `intellij-instance-2022.2.1--T3ySdgShSvyr87HNoJq-oQ/Contents`

Reply:
Wow okay 🍎 pls document in code comment

Reply:
Also, isn't the _version number_ just an internal implementation detail? as in, do we care whether the path contains the version or not, as long as it point to a valid IDE dir?

Reply:
Hm, if there would be multiple tests running in one process on the same instance (machine / docker instance etc.) and these different tests would use different versions of intellij - it would be nice to have logs grouped by intellij versions at least. That's why I decided o take it into account. Does it make sense? Or we need some other logic for such scenarios?

Reply:
Ok, makes sense